File: standardization/Algorithms/SBR-JAVA/src/main/java/Node.java
import java.util.ArrayList;
import java.util.List;

/**
 * Created by ran on 12.8.2017.
 *
 */
public abstract class Node {

    protected int ID;

    protected String label;

    protected boolean isRoot;

    protected boolean isComplete = false;

    protected List<Integer> tags = new ArrayList<Integer>();

    protected List<Integer> seqOf = new ArrayList<Integer>();

    protected List<Integer> seq = new ArrayList<Integer>();

    /****************************************
     * method: root
     * input:
     * output: boolean isroot.
     * operation: return True if this is root, false otherwise.
     ****************************************/
    public boolean root() {
        return isRoot;
    }

    /****************************************
     * method: tag
     * input:
     * 		int n - time_stamp.
     * 		String s - 'soft' or 'hard', depends on kind of tag.
     * output:
     * operation: checks string, and assign tag accordingly.
     ****************************************/
    public void tag(int n) {
        tags.add(n);
    }

    /****************************************
     * method: delete_tag
     * input:
     * 		Integer n - time_stamp to remove from tags.
     * output:
     * operation: removes according tag.
     ****************************************/
    public void delete_tag(Integer n) {
        tags.remove(n);
    }

    /****************************************
     * method: tagged
     * input:
     * 		Integer n - time_stamp.
     * 		String s - 'soft' or 'hard'.
     * output:
     * 		boolean value - True if this is tagged, False accordingly.
     * operation: checks given String, then checks according list to find a match to time_stamp.
     ****************************************/
    public boolean tagged(Integer n) {
        return tags.contains(n);
    }

    /****************************************
     * method: NoSeqEdges
     * input:
     * output:
     * 		boolean value - True if no sequential edges exist, False otherwise.
     * operation: checks if there are sequential edges.
     ****************************************/
    public boolean NoSeqEdges() {
        return this.seqOf.isEmpty();
    }

    public void addSeqOf(int ID, Node p) {
        this.seqOf.add(ID);
        p.seq.add(this.ID);
    }

    /****************************************
     * method: getLabel
     * input:
     * output:
     * 		String label.
     * operation: getter.
     ****************************************/
    public String getLabel() {
        return label;
    }

    public void setRoot(boolean isRoot)
    {
        this.isRoot = isRoot;
    }

    public List<Integer> getSeq() {
        return seq;
    }

    public List<Integer> getSeqOf() {
        return seqOf;
    }

    public void setComplete(boolean complete) {
        isComplete = complete;
    }

    public boolean getIsComplete() {return isComplete;}

    public List<Integer> getTags()
    {
        return tags;
    }

    public int getID()
    {
        return ID;
    }
}

File: standardization/Algorithms/SBR/Node.py
class Node(object):

    def __init__(self, ID ,label, is_root , is_complete, tags, seq_of, seq):
        self._label = label
        self._is_root = is_root
        self._id = ID
        self._is_complete = is_complete
        self._tags = tags
        self._seq_of = seq_of
        self._seq = seq

    def root(self):
        return self._is_root

    def tag(self, t):
        self._tags.append(t)

    def delete_tag(self, t):
        self._tags.remove(t)

    def tagged(self, t):
        return t in self._tags

    def no_seq_edges(self):
        return self._seq_of == []

    def add_seq_of(self, ID, p):
        self._seq_of.append(ID)
        p._seq.append(self._id)

    def get_label(self):
        return self._label

    def set_root(self, is_root):
        self._is_root = is_root

    def get_seq(self):
        return self._seq

    def get_seq_of(self):
        return self._seq_of

    def set_complete(self, complete):
        self._is_complete = complete

    def get_is_complete(self):
        return self._is_complete

    def get_tags(self):
        return self._tags

    def get_ID(self):
        return self._id

    def get_root(self):
        return self._is_root

    def set_ID(self, ID):
        self._id = ID